src/services/package-manager/batch-import-export-service.ts

```diff
@@ -0,0 +1,59 @@
+import {batchImportExportApi} from "../../api/batch-import-export-api";
+import {logger} from "../../util/logger";
+import {v4 as uuidv4} from "uuid";
+import {PackageExportTransport} from "../../interfaces/package-export-transport";
+import {FileService, fileService} from "../file-service";
+import {packageApi} from "../../api/package-api";
+import {variableService} from "./variable-service";
+import {PackageManagerVariableType} from "../../interfaces/package-manager.interfaces";
+import {dataModelService} from "./datamodel-service";
+
+class BatchImportExportService {
+
+    public async listActivePackages(flavors: string[]): Promise<void> {
+        const activePackages = await batchImportExportApi.findAllActivePackages(flavors);
+        activePackages.forEach(pkg => {
+            logger.info(`${pkg.name} - Key: "${pkg.key}"`)
+        });
+    }
+
+    public async findAndExportListOfActivePackages(flavors: string[], packageKeys: string[], withDependencies: boolean): Promise<void> {
+        let packagesToExport: PackageExportTransport[];
+
+        if (packageKeys.length) {
+            packagesToExport = await batchImportExportApi.findActivePackagesByKeys(packageKeys, withDependencies);
+        } else  {
+            packagesToExport = await batchImportExportApi.findAllActivePackages(flavors, withDependencies);
+        }
+
+        const packageByKey = new Map<string, PackageExportTransport>();
+        packagesToExport.forEach(pkg => packageByKey.set(pkg.key, pkg));
+
+        if (withDependencies) {
+            const studioPackageWithDataModelVariableAssignments = await variableService.getVariableAssignmentsForNodes(PackageManagerVariableType.DATA_MODEL);
+            const dataModelDetailsByNode = await dataModelService.getDataModelDetailsForPackages(studioPackageWithDataModelVariableAssignments);
+            studioPackageWithDataModelVariableAssignments.forEach(studioPackage => {
```

Comment:
Do you think it's better if we extract these custom studio logic into a new service, and handle everything there? I think it would help us to keep the flow clean and to easily see what is being done for studio purposes.